azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -398,10 +403,88 @@ public void createContainer(final int executionId) throws ExecutorManagerExcepti
    */
   @Override
   public void deleteContainer(final int executionId) throws ExecutorManagerException {
-    deletePod(executionId);
-    if (isServiceRequired()) {
-      deleteService(executionId);
+    try { // if pod deletion is not successful, the service deletion can still be handled
+      deletePod(executionId);
+    } finally {
+      if (isServiceRequired()) {
+        deleteService(executionId);
+      }
+    }
+  }
+
+  /**
+   * This method is used to delete invalid containers in batch
+   * @param containerValidity container valid duration
+   * @throws ExecutorManagerException
+   */
+  @Override
+  public void deleteAgedContainers(final Duration containerValidity) throws ExecutorManagerException {
+    logger.info(String.format("Cleaning up containers older than %d days",
+        containerValidity.toDays()));
+    final Set<Integer> containersToDelete = getStaleContainers(containerValidity);
+    for (final int execId : containersToDelete) {
+      if (execId > 0) {
+        logger.info(String.format("Deleting stale pod %s and service %s", getPodName(execId),
+            getServiceName(execId)));
+        try {
+          deleteContainer(execId);
+        } catch (ExecutorManagerException e) {
+          logger.error(String.format("Unable to delete stale pod and service of exec-id %d",
+              execId), e.getMessage());
+        }
+      }
+    }
+  }
+
+  /**
+   * This method is used to fetch all stale pods in the current az cluster and namespace
+   * @param containerValidity container valid duration
+   * @return Set of stale container's execution ids
+   */
+  private Set<Integer> getStaleContainers(final Duration containerValidity) throws ExecutorManagerException {
+    try {
+      // Select pods only from current Azkaban cluster and namespace
+      final String label = CLUSTER_LABEL_NAME + "=" + this.clusterName + "," + LABEL_POSTFIX;
```

Comment:
please, consider using `APP_LABEL_NAME + "=" + POD_APPLICATION_TAG` instead of defining constant `LABEL_POSTFIX`

Reply:
Used APP_LABEL_NAME + "=" + POD_APPLICATION_TAG instead of LABEL_POSTFIX